de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaDataSubGenerator.xtend

```diff
@@ -0,0 +1,197 @@
+ */
+class LemmaDataSubGenerator {
+    /**
+     * Map of all created DataStructures during a generation. The key contains the full-qualified
+     * name while the value contains the actual data structure created.
+     */
+    val createdDataStructures = <String, DataStructure>newHashMap
+    /**
+     * Map of all created Lists during a generation. The key contains the full-qualified
+     * name while the value contains the actual list structure created.
+     */
+    val createdListStructures = <String, ListType>newHashMap
+    /** Factory to actually create and manipulate a LEMMA DataModel */
+    val DATA_FACTORY = DataFactory.eINSTANCE
+    /** Predefined instances of the data model, version, and context
+     * which are enriched with the data structures from the OpenAPI source
+     */
+    val myDataModel = DATA_FACTORY.createDataModel
+    val myVersion = DATA_FACTORY.createVersion
+    val myContext = DATA_FACTORY.createContext
+    /** Seperator is used to build the full qualified names during the generation */
+    val seperator = "."
+    /** OpenAPI schema which will be used as source for generation */
+    val OpenAPI openAPI
+    /** SLF4j Logger */
+    val static logger = LoggerFactory.getLogger(LemmaDataSubGenerator)
+    /** Location where the generated file is written */
+    val String targetFolder
+
+    new(OpenAPI openAPI, String genPath, String dataFilename) {
+        super()
+        logger.debug("Creating new Data Sub Generator...")
+        this.openAPI = openAPI
+        this.targetFolder = genPath + dataFilename
+    }
+
+    def DataModel generate() {
+        logger.debug("Starting generation...")
+        logger.debug("Initializing model instance...")
+        initilize()
+        logger.debug("...data model initialized!")
+        logger.debug("Creating data structures...")
+
+        openAPI?.components?.schemas?.forEach[
+            key, value| getOrCreateDataStructure(myContext, key, value)
+        ]
+        logger.debug("...data structures created!")
+
+        if (OpenApiUtil.writeModel(myDataModel, targetFolder)) {
+            logger.info("Data model generation successful!")
+            logger.info('''Model written to «targetFolder»''')
+        } else
+            throw new Exception("Data model generation failed :(")
+        return myDataModel
+    }
+
+    def initilize() {
+        // Data Model
+        myDataModel.versions.add(myVersion)
+        // Version
+        myVersion.name = OpenApiUtil.removeInvalidCharsFromName(openAPI.info.version)
+        myVersion.contexts.add(myContext)
+        // Context
+        myContext.name = OpenApiUtil.removeInvalidCharsFromName(openAPI.info.title)
+        myContext.version = myVersion
+    }
+
+    def DataStructure getOrCreateDataStructure(
+        Context c, String name, Schema schema
+    ) {
+        var foundObject = findDataStructure(c, StringUtils.capitalize(name))
```

Comment:
Order the methods in your code in the order of the calls. `findDataStructure` should immediately follow `getOrCreateDataStructure` and so forth. This improves "reviewability" and readability by a lot if you don't use an IDE.